jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunSettingsEditorPanel.java

```diff
@@ -71,28 +80,44 @@ private void createUIComponents() {
         functionModels = new DefaultComboBoxModel<>();
         function = new ComboBox<>(functionModels);
         function.addActionListener(e -> updateComponents());
+
+        environmentVariables = new ResourceEnvironmentVariablesField();
+        Disposer.register(this, environmentVariables);
+    }
+
+    private ResourceEnvironmentVariablesField getEnvironmentVariablesFieldImpl() {
+        return (ResourceEnvironmentVariablesField) environmentVariables;
+    }
+
+    private void acceptResourceMapping(ResourceMapping mapping) {
+        if (!Disposer.isDisposed(this)) {
+            getEnvironmentVariablesFieldImpl().setResourceMapping(mapping);
+        }
     }
 
     private void updateComponents() {
         handler.setEnabled(!useTemplate.isSelected());
         runtime.setEnabled(!useTemplate.isSelected());
         templateFile.setEnabled(useTemplate.isSelected());
 
+        Function selectedFunction = null;
         if (useTemplate.isSelected()) {
             handler.setBackground(UIUtil.getComboBoxDisabledBackground());
             handler.setForeground(UIUtil.getComboBoxDisabledForeground());
 
             if (functionModels.getSelectedItem() instanceof Function) {
-                Function selected = (Function) functionModels.getSelectedItem();
-                handler.setText(selected.handler());
-                runtime.setSelectedItem(Runtime.fromValue(selected.runtime()));
+                selectedFunction = (Function) functionModels.getSelectedItem();
```

Comment:
Can we push this logic into a function/method and return the `function` instance so that we don't have to declare the `Function selectedFunction = null` call above?